neo/Prompt/Commands/tests/test_send_command.py

```diff
@@ -328,25 +326,15 @@ def test_could_not_send(self, mocked_tracback_module):
         self.assertFalse(res)
 
     def test_sendmany_good_simple(self):
-        with patch('neo.Prompt.Commands.Send.prompt', side_effect=["neo", self.watch_addr_str, "1", "gas", self.watch_addr_str, "1", UserWalletTestCase.wallet_1_pass()]):
-
-            wallet = self.GetWallet1(recreate=True)
-            args = ["2"]
-
-            framework = construct_send_many(wallet, args)
-            res = process_transaction(wallet, contract_tx=framework[0], scripthash_from=framework[1], scripthash_change=framework[2], fee=framework[3], owners=framework[4], user_tx_attributes=framework[5])
-
-            self.assertTrue(res)  # verify successful tx
-
-            json_res = res.ToJson()
-            self.assertEqual(self.watch_addr_str, json_res['vout'][0]['address'])  # verify correct address_to
+        wallet = self.GetWallet1(recreate=True)
+        with patch('neo.Prompt.PromptData.PromptData.Wallet', return_value=wallet):
+            with patch('neo.Prompt.Commands.Send.prompt', side_effect=["neo", self.watch_addr_str, "1", "gas", self.watch_addr_str, "1", UserWalletTestCase.wallet_1_pass()]):
+                with patch('sys.stdout', new=StringIO()) as mock_print:
+                    args = ['sendmany', '2']
+                    command_wallet = Wallet.CommandWallet()
+                    command_wallet.execute(args)
 
-            # check for 2 transfers
-            transfers = 0
-            for info in json_res['vout']:
-                if info['address'] == self.watch_addr_str:
-                    transfers += 1
-            self.assertEqual(2, transfers)
+                    self.assertIn("Relayed Tx", mock_print.getvalue())
```

Comment:
the `execute()` method does not have an enforced return type. Use this to your advantage to return the normal `res` value from `process_transaction` as in the original tests
[here](https://github.com/CityOfZion/neo-python/pull/726/files?utf8=%E2%9C%93&diff=split#diff-d795c43528b80a2b124aeb31f31808afR51)

this way you don't have to patch `stdout` and you have far more options for testing correct behaviour. We now threw away good logic in favour of trusting a printed message indicating all went right. 

Note that the actual goal isn't coverage. The goal is testing correct behaviour of the functions/modules under different circumstances. `coverage` is just an indicator that can tell you if you missed any obvious parts in your test scenario's

Reply:
I tried this originally and all it ever returned was `None`. Also `pylint` warned about it in VSC. Do you have any hints?

This is why I moved the test logic to the next test. So, this basic test shows everything works and then I test the original functions thoroughly because they return values other than `None`

Reply:
@LysanderGG if you read my comment above you see I assumed we could use the return value of `execute()`. Atm this doesn't work because for the `CommandBase.execute_sub_command` doesn't return the value. Do you see/have any objections for changing the behaviour of that? I don't have a full clear mind at this point to see possible limitations and you know this code the best :)

If we return the results of `execute` commands they become more obvious to test. The other approach is just directly calling a the command class. An example test of the former is:
```
                    args = ['send', 'neo', self.watch_addr_str, '50']                    
                    command_wallet = Wallet.CommandWallet()
                    res = command_wallet.execute(args)
``` 
an example of the latter is
```
                    args = ['neo', self.watch_addr_str, '50']                    
                    command_wallet_send = Wallet.CommandWalletSend()
                    res = command_wallet_send.execute(args)
```